augur/filter/include_exclude_rules.py

```diff
@@ -201,18 +201,27 @@ def filter_by_query(metadata: pd.DataFrame, query: str, column_types: Optional[D
         # Column extraction failed. Apply type conversion to all columns.
         columns = metadata_copy.columns
 
-    # If a type is not explicitly provided, try converting the column to numeric.
-    # pd.to_numeric supports nullable numeric columns unlike pd.read_csv's
-    # built-in data type inference.
+    # If a type is not explicitly provided, try automatic conversion.
     for column in columns:
-        column_types.setdefault(column, 'numeric')
+        column_types.setdefault(column, 'auto')
 
     # Convert data types before applying the query.
     # NOTE: This can behave differently between different chunks of metadata,
     # but it's the best we can do.
     for column, dtype in column_types.items():
-        if dtype == 'numeric':
```

Comment:
The `numeric` dtype is still an accepted type for `--query-columns` so it should still probably still have it's own `elif` block here?

Reply:
Thanks for the sharp-eyed review! [Fixed](https://github.com/nextstrain/augur/compare/6765b8bbbc46cd109edb7b08bc4e02e35564a46d..d44506da4544b0f52caebd77dc565031165b33f1).

Since `--query-columns` is unreleased, I thought it'd be best to remove support for `numeric` entirely: 570f5639

I've also added `bool` to the `ACCEPTED_TYPES`.

Reply:
Ah, didn't realize it wasn't released yet. Makes sense to just remove!